legacy/src/main/java/org/opensearch/sql/legacy/esdomain/LocalClusterState.java

```diff
@@ -79,25 +64,30 @@ public static synchronized void state(LocalClusterState instance) {
     INSTANCE = instance;
   }
 
-  public void setClusterService(ClusterService clusterService) {
+  /**
+   * Sets the ClusterService used to receive ClusterSetting update notifications.
+   * @param clusterService The non-null cluster service instance.
+   */
+  public void setClusterService(@NonNull ClusterService clusterService) {
     this.clusterService = clusterService;
+  }
 
-    clusterService.addListener(
-        event -> {
-          if (event.metadataChanged()) {
-            // State in cluster service is already changed to event.state() before listener fired
-            if (LOG.isDebugEnabled()) {
-              LOG.debug(
-                  "Metadata in cluster state changed: {}",
-                  new IndexMappings(clusterService.state().metadata()));
-            }
-            cache.invalidateAll();
-          }
-        });
+  /**
+   * Sets the Client used to interact with OpenSearch core.
+   * @param client The non-null client instance
+   */
+  public void setClient(@NonNull Client client) {
+    this.client = client;
   }
 
-  public void setPluginSettings(OpenSearchSettings settings) {
+  /**
+   * Sets the plugin's settings.
+   * @param settings The non-null plugin settings instance
+   */
+  public void setPluginSettings(@NonNull OpenSearchSettings settings) {
```

Comment:
Slightly outside the scope of this PR, but it's not immediately obvious why this class is registering a settings update consumer and maintaining it's local cache of settings in `latestSettings`. It appears to be duplicating what is already happening in OpenSearchSettings. I wonder if `getSettingValue()` can be changed to just be a pass-through to OpenSearchSettings and all usage of the ClusterService can be removed from this class?

Reply:
I get the feeling that you're right, but I also haven't spent enough time around it to be sure